### PR TITLE
review: fix: some positions bugs

### PR DESCRIPTION
**Pull request by pvojtechovsky**

### CODE REVIEW

**Comment by tdurieux** [comment]
Looks good for me

File: src/main/java/spoon/support/compiler/jdt/PositionBuilder.java [hunk 1]
```diff
@@ -170,24 +198,82 @@ SourcePosition buildPositionCtElement(CtElement e, ASTNode node) {
...
+				return off;
 			}
+			off++;
 		}
+		return -1;
+	}
 
-		return cf.createSourcePosition(cu, sourceStart, sourceEnd, lineSeparatorPositions);
+	/**
+	 * @return index of first whitespace char, searching backward. Can return off if it is whitespace.
```
Comment (surli) [inline src/main/java/spoon/support/compiler/jdt/PositionBuilder.java]: Wrong doc: must be "forward" for this one.

File: src/main/java/spoon/support/compiler/jdt/PositionBuilder.java [hunk 2]
```diff
@@ -170,24 +198,82 @@ SourcePosition buildPositionCtElement(CtElement e, ASTNode node) {
...
-			int statementEnd = expression.statementEnd;
 
-			if (statementEnd > 0) {
-				sourceEnd = statementEnd;
+		return cf.createSourcePosition(cu, sourceStart, sourceEnd, lineSeparatorPositions);
+	}
+
+	/**
+	 * @return index of first non whitespace char, searching backward. Can return off if it is non whitespace.
+	 */
```
Comment (surli) [inline src/main/java/spoon/support/compiler/jdt/PositionBuilder.java]: Same here: must be forward in doc.

**Comment by pvojtechovsky** [comment]
> Wrong doc: must be "forward" for this one.

Thanks for pointing to that. I fixed that.

**Pull request merged**